### DISCUSSION TITLE
How is ZydisDecodedOperand.imm.is_signed determined?

### DISCUSSION THREAD

**Discussion by McSinyx** [zyantific/zydis#538] [Q&A] [answered]
When Zydis encounter an immediate like `$0x8877665544332211`, what hint can be given to mark it as signed or is it determined based on the instruction?

**Comment by mappzor** [comment, marked as answer]
Every instruction definition contains information about physical encoding of its operands. During decoding that information is extracted from the definition and passed to the user as `ZydisDecodedOperand.encoding`. See [docs](https://doc.zydis.re/master/html/SharedTypes_8h#a8e8832e13c91d098d8b7288a00aba02b) for possible values.

**Reply by McSinyx** [reply]
Thanks!
